File: packages/python-sdk/src/blindfold/regex/detectors/us/passport.py
"""US Passport number detector (context-required)"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region


@register_region("us")
class UsPassportDetector(RegexDetector):
    entity_type = EntityType.US_PASSPORT
    score = 0.75
    context_keywords = [
        "passport", "passport#", "passport #", "passport number",
        "passport no",
    ]
    context_required = True
    context_window = 50

    # US passport: 9 digits
    pattern = re.compile(r"\b\d{9}\b")

File: packages/js-sdk/src/regex/detectors/us/passport.ts
// US Passport number detector (context-required)

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'

class UsPassportDetector extends RegexDetector {
  entityType = EntityType.US_PASSPORT
  score = 0.75
  contextKeywords = ['passport', 'passport#', 'passport #', 'passport number', 'passport no']
  contextRequired = true
  contextWindow = 50

  // US passport: 9 digits
  pattern = /\b\d{9}\b/g
}

registerRegion('us', UsPassportDetector)

export { UsPassportDetector }
